### PR TITLE
Experiment/publish workflow

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Description**
- Experimenting with formalizing the versioning and publishing process.
   - use 'npm run release' to do a real publish / release
   - use 'npm run release-beta' for a beta publish
- Sorry the diff sucks; needed to format package.json so it's consistent with all of our other formatting
- cc @tfennelly @scherler @kzantow @imeredith @sophistifunk @michaelneale 

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by cliffmeyers** [comment]
I haven't tested the real "npm run release" yet but am hoping I interpreted the docs correctly and it will just lop off the prelease part without ticking up the version. When I have a little more time I will fiddle with it and test it some more.

**Comment by sophistifunk** [comment]
I like it, particularly that it will add a tag to git. 🐝

**Comment by cliffmeyers** [comment]
@tfennelly curious to hear your feedback since this adjusts slightly how the "site-publish" feature is called.

**Comment by kzantow** [comment]
FYI, you can see the "real" diff this way: https://github.com/jenkinsci/jenkins-design-language/pull/94/files?w=1

File: package.json [hunk 1]
```diff
@@ -1,88 +1,91 @@
...
+    "storybook": "npm run gulp build && start-storybook -p 9001 --static-dir dist/assets/",
+    "build-storybook": "build-storybook --static-dir dist/assets/ --output-dir website/storybook",
+    "flow": "flow",
+    "gulp": "gulp",
+    "test": "gulp test",
+    "site-build": "harp compile website/ website-build/",
+    "site-server": "harp server website --port 3000",
+    "site-publish": "./site-publish.sh",
+    "preversion": "npm install && gulp",
+    "release": "npm version patch && npm publish && npm run site-publish",
```
Comment (kzantow) [inline package.json]: Will this actually run a build/test?

Comment (cliffmeyers) [inline package.json]: Yes. npm version will call "preversion" and "gulp" with no args runs a full lint / test / build.

Comment (kzantow) [inline package.json]: Good to know about `preversion` :)

**Comment by cliffmeyers** [comment]
@tfennelly have any feedback on these changes?

**Comment by cliffmeyers** [comment]
Going to close this PR in favor of #96. Wanted a cleaner commit history without the numerous tags and other junk floating in this. The notable differences in the new PR are the addition of "validate-deps" in package.json scripts... and the validator node script itself.

cc @kzantow @sophistifunk @tfennelly

**Pull request closed without merging**